lib/core/utils/shadow-select.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Find the first element to match a selector.
+ * Use an array of selectors to reach into shadow DOM trees
+ * 
+ * @param {string|string[]} selector String or array of strings with a CSS selector
+ * @param {Document} doc Optional document node
+ * @returns {Element|Null}
+ */
+export default function shadowSelect(selectors) {
```

Comment:
So this function has a very specific and singular purpose: to select frame elements returned from `getFrameContexts`. The frames should be selectable without the virtual tree and using the selector `getFrameContexts` returns. I think we should rename this function to be very specific as to when / why to use it (and that it works specifically in conjunction with `getFrameContexts`).`getFrameInContext` perhaps?

Also, this function cannot be used by highlighter or directly with axe result selectors since those can include iframe elements which this does not handle (e.g. `["iframe", ["#shadow", "button"]]`).

```suggestion
export default function getFrameInContext(selectors) {
```

Reply:
There is nothing in this function that makes it just about frames. This is not a good name.